build_steps/docker-compose.yml

```diff
@@ -0,0 +1,28 @@
+version: "3.4"
```

Comment:
I don't quite see why docker compose is required....I thought it was intended to combine together multiple containers whereas, couldn't what you do here be achieved with a single ``docker run`` incantation?

Also, more fundamentally, the way I imagine the GeNN docker image being used is that it would be hosted on Dockerhub (probably produced by CI when we make releases/merge master) so users wouldn't have to checkout our git repro and hence wouldn't have access to the docker compose script outside of the container.

Again, I may be missing things....

Reply:
Agreed, it facilitates having multiple containers interacting together. I only used it as I find it helpful to set up port bindings, volumes, health checks, etc in a configurable way that is easy to use (and was used only to launch jupyter).

However, if it is not going to be useful for extendability then maybe, as you say, a command such as `docker run -p 127.0.0.1:8888:8888/tcp -v '$(CURDIR)'/pygenn/notebooks:/root/pygenn/notebooks pygenn jupyter notebook --ip 0.0.0.0 --no-browser` is better suited. I have added this to the makefile (as `make docker-jupyter`) and can delete the `docker-compose.yml`?

Reply:
Just for saving such brutally long command lines, docker compose seems like a good idea (especially as it's optional).